com.unity.formats.alembic/Editor/Importer/AlembicImporter.cs

```diff
@@ -20,6 +19,22 @@
 [assembly: InternalsVisibleTo("Unity.Formats.Alembic.UnitTests.Editor")]
 namespace UnityEditor.Formats.Alembic.Importer
 {
+    class AlembicAssetPostProcessor : AssetPostprocessor
+    {
+#if UNITY_2021_3_OR_NEWER
+        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths, bool didDomainReload)
+#else
+        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
+#endif
+        {
+            foreach (var path in movedAssets)
+            {
+                if (path.EndsWith(".abc", StringComparison.OrdinalIgnoreCase))
+                    AssetDatabase.ImportAsset(path);
```

Comment:
Could it be a bug in the asset database? It feels strange to have to do that ourselves.
Is the Alembic importer that odd? Or is it a special case?

Reply:
It's a specific from our Alembic plugin. We store a piece of data in one of the Alembic SubAsset that needs to be updated if the Alembic file is moved. The ADB is doing its job as expected. We're actually missing a proper way to handle this case.

I'm trying to figure if there's a better way to handle this but at the current state of the fix, I would simply trigger a reimport of the asset. Without this PR, the fix/workaround for users is to also `Right click > Reimport` on the affected assets.

Reply:
I thought sub-assets would be re-imported as well as the main asset when you move it. Actually, if the main asset is correctly re-imported when it's moved, it should "re-build" the sub-assets no? As they're part of the importer (in a normal case at least).

If the sub-asset is not generated by the importer, but later... can't there be at least a dependency between the two? (I'm not entirely sure how that work, maybe it doesn't make sense, I just know you can set dependencies between imported assets so that when one is re-imported the other is as well).

(we can do a quick huddle if it's simpler, I'm curious to understand the case so that I also better understand how the asset database works)

Reply:
I pushed an update in [a8c902c](https://github.com/Unity-Technologies/com.unity.formats.alembic/pull/548/commits/a8c902c5f1bf5e96a926b148ec18109b058381a2).
It sets a dependency with the assetPath more nicely. No more tricks!